src/sinks/datadog/metrics/encoder.rs

```diff
@@ -738,6 +760,47 @@ mod tests {
         }
     }
 
+    fn encode_sketches_normal<B>(
+        metrics: &[Metric],
+        default_namespace: &Option<Arc<str>>,
+        log_schema: &'static LogSchema,
+        buf: &mut B,
+    ) where
+        B: BufMut,
+    {
+        let mut sketches = Vec::new();
+        for metric in metrics {
+            match metric.value() {
+                MetricValue::Sketch { sketch } => match sketch {
+                    MetricSketch::AgentDDSketch(ddsketch) => {
+                        // Don't encode any empty sketches.
+                        if ddsketch.is_empty() {
+                            continue;
+                        }
+
+                        let sketch = sketch_to_proto_message(
+                            metric,
+                            ddsketch,
+                            default_namespace,
+                            log_schema,
+                        );
+
+                        sketches.push(sketch);
+                    }
+                },
+                _ => panic!("bad"),
+            }
+        }
+
+        let sketch_payload = ddmetric_proto::SketchPayload {
+            metadata: None,
```

Comment:
Is there any specific reason why metadata is never populated?

Reply:
So, this is the old, now-removed comment about this field when we were constructing the `Sketch` payload all at once:

```
// TODO: The "common metadata" fields are things that only very loosely apply to Vector, or
// are hard to characterize -- for example, what's the API key for a sketch that didn't originate
// from the Datadog Agent? -- so we're just omitting it here in the hopes it doesn't
// actually matter.
```

`CommonMetadata` has stuff like public IP, internal IP, timezone, agent version, and API key. We don't have most of those at all, and for stuff like API key, we're already specifying it at a higher level in the request to the intake API. So we've never actually done anything with it.

Reply:
Got it, thanks!